datajoint/base_relation.py

```diff
@@ -142,8 +142,19 @@ def insert(self, rows, replace=False, ignore_errors=False, skip_duplicates=False
         """
 
         if isinstance(rows, RelationalOperand):
-            # INSERT FROM SELECT
-            query = 'INSERT{ignore} INTO {table} ({fields}) {select}'.format(
+            # INSERT FROM SELECT - build alternate field-narrowing query (only) when needed
+            if ignore_extra_fields and False in [name in self.heading.names for name in rows.heading.names]:
+                alias = '___' + self.table_name + '_IFS_' + rows.table_name
```

Comment:
Be aware, that `table_name` is *not* a property of a `RelationalOperand`. This line would only work if the `rows` happen to be a `BaseRelation` which is not a requirement for `select from` syntax.